Exception while decoding a transaction envelope containing a custom asset operation

```  
    //This works.
    @Test 
    public void decodeTransactionEnvelopeWithNativeAsset() throws IOException {

        String transactionEnvelopeToDecode = "AAAAACq1Ixcw1fchtF5aLTSw1zaYAYjb3WbBRd4jqYJKThB9AAAAZAA8tDoAAAAEAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAAHMXLrCwa9tH+ABeAAPp1pgmHqBS9e0879ylTSHCrChUAAAAAAAAAAB3NZQAAAAAAAAAAAA==";
        Base64 base64Codec = new Base64();

        byte[] bytes = base64Codec.decode(transactionEnvelopeToDecode);

        TransactionEnvelope transactionEnvelope = TransactionEnvelope.decode(new XdrDataInputStream(new ByteArrayInputStream(bytes)));
        //no exceptions thrown here :)

        org.stellar.sdk.xdr.Transaction tx = transactionEnvelope.getTx();

        assertEquals(100, tx.getFee().getUint32().intValue());
        //this works.Yoopie!

    }

    //This doesn't work.
    @Test 
    public void decodeTransactionEnvelopeWithCustomAsset() throws IOException {

        String transactionEnvelopeToDecode = "AAAAACq1Ixcw1fchtF5aLTSw1zaYAYjb3WbBRd4jqYJKThB9AAAAZAA8tDoAAAALAAAAAAAAAAEAAAAZR29sZCBwYXltZW50IGZvciBzZXJ2aWNlcwAAAAAAAAEAAAAAAAAAAQAAAAARREGslec48mbJJygIwZoLvRtL6/gGL4ss2TOpnOUOhgAAAAFHT0xEAAAAACq1Ixcw1fchtF5aLTSw1zaYAYjb3WbBRd4jqYJKThB9AAAAADuaygAAAAAAAAAAAA==";

        Base64 base64Codec = new Base64();

        byte[] bytes = base64Codec.decode(transactionEnvelopeToDecode);

        TransactionEnvelope transactionEnvelope = TransactionEnvelope.decode(new XdrDataInputStream(new ByteArrayInputStream(bytes)));
        //TransactionEnvelope.decode is throwing the following exception :(
//        **Caused by: java.lang.RuntimeException: Unknown enum value: 21450572**
//	at org.stellar.sdk.xdr.AssetType.decode(AssetType.java:41) ~[stellar-1.jar:0.1.12]
//	at org.stellar.sdk.xdr.Asset.decode(Asset.java:72) ~[stellar-1.jar:0.1.12]
//	at org.stellar.sdk.xdr.PaymentOp.decode(PaymentOp.java:50) ~[stellar-1.jar:0.1.12]
//	at org.stellar.sdk.xdr.Operation$OperationBody.decode(Operation.java:207) ~[stellar-1.jar:0.1.12]
//	at org.stellar.sdk.xdr.Operation.decode(Operation.java:78) ~[stellar-1.jar:0.1.12]
//	at org.stellar.sdk.xdr.Transaction.decode(Transaction.java:121) ~[stellar-1.jar:0.1.12]
//	at org.stellar.sdk.xdr.TransactionEnvelope.decode(TransactionEnvelope.java:47) ~[stellar-1.jar:0.1.12]

        //unreachable code below.
        org.stellar.sdk.xdr.Transaction tx = transactionEnvelope.getTx();

        assertEquals(100, tx.getFee().getUint32().intValue());

    }
```

--- a/src/main/java/org/stellar/sdk/xdr/XdrDataInputStream.java
+++ b/src/main/java/org/stellar/sdk/xdr/XdrDataInputStream.java
@@ -119,7 +119,7 @@ public class XdrDataInputStream extends DataInputStream {
             }
 
             while (pad-- > 0) {
-                int b = mIn.read();
+                int b = read();
                 if (b != 0) {
                     throw new IOException("non-zero padding");
                 }

--- a/src/test/java/org/stellar/sdk/xdr/TransactionDecodeTest.java
+++ b/src/test/java/org/stellar/sdk/xdr/TransactionDecodeTest.java
@@ -5,8 +5,10 @@ import org.junit.Test;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.util.Arrays;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class TransactionDecodeTest {
 
@@ -44,6 +46,17 @@ public class TransactionDecodeTest {
 
         TransactionMeta transactionMeta = TransactionMeta.decode(new XdrDataInputStream(new ByteArrayInputStream(bytes)));
         assertEquals(1, transactionMeta.getOperations().length);
+    }
+
+    @Test
+    public void testTransactionEnvelopeWithMemo() throws IOException {
+        String transactionEnvelopeToDecode = "AAAAACq1Ixcw1fchtF5aLTSw1zaYAYjb3WbBRd4jqYJKThB9AAAAZAA8tDoAAAALAAAAAAAAAAEAAAAZR29sZCBwYXltZW50IGZvciBzZXJ2aWNlcwAAAAAAAAEAAAAAAAAAAQAAAAARREGslec48mbJJygIwZoLvRtL6/gGL4ss2TOpnOUOhgAAAAFHT0xEAAAAACq1Ixcw1fchtF5aLTSw1zaYAYjb3WbBRd4jqYJKThB9AAAAADuaygAAAAAAAAAAAA==";
+        Base64 base64Codec = new Base64();
+        byte[] bytes = base64Codec.decode(transactionEnvelopeToDecode);
+
+        TransactionEnvelope transactionEnvelope = TransactionEnvelope.decode(new XdrDataInputStream(new ByteArrayInputStream(bytes)));
+        assertEquals(1, transactionEnvelope.getTx().getOperations().length);
+        assertTrue(Arrays.equals(new byte[]{'G', 'O', 'L', 'D'}, transactionEnvelope.getTx().getOperations()[0].getBody().getPaymentOp().getAsset().getAlphaNum4().getAssetCode()));
     }
 
 }

